src/wasm/wasm-ir-builder.cpp

```diff
@@ -933,6 +944,10 @@ Result<> IRBuilder::visitElse() {
   if (binaryPos && func) {
     func->delimiterLocations[iff][BinaryLocations::Else] =
       lastBinaryPos - codeSectionOffset;
+
+    // Note the start of the if (which will be lost as the If is closed and the
+    // Else begins, but the if spans them both).
+    func->expressionLocations[iff].start = scope.startPos - codeSectionOffset;
```

Comment:
Can we either fix this for `try-catch` as well, or add a TODO to do so?

Reply:
Added a TODO. I'm not sure offhand how to handle multiple catches. Anyhow, only If is urgent atm.